src/plugins/intel_gpu/src/graph/tile.cpp

```diff
@@ -17,8 +17,6 @@ namespace cldnn {
 GPU_DEFINE_PRIMITIVE_TYPE_ID(tile)
 
 layout tile_inst::calc_output_layout(tile_node const& node, kernel_impl_params const& impl_param) {
-    assert(static_cast<bool>(impl_param.desc->output_data_types[0]) == false &&
-           "Output data type forcing is not supported for tile_node!");
```

Comment:
[Random spot] Have you considered overriding output layout data type somewhere here https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_gpu/src/graph/include/primitive_type_base.h#L119 in case if it's available? Likely it will also help to find the issues with incorrect type passed from Op converters mentioned in the other comment, as looks like currently most of calc_output_layout(s) impls just ignore those optional data types.

Reply:
I've modified ops/transpose to set the output data-type. It needs to update graph/permute only.
Common logic to check the output dt seems unnecessary.